packages/sdk/wallets/wallet-walletconnect/src/wc-signer.ts

```diff
@@ -0,0 +1,91 @@
+import { CeloTx, EncodedTransaction, Signer } from '@celo/connect'
+import { EIP712TypedData } from '@celo/utils/src/sign-typed-data-utils'
+import WalletConnect from '@walletconnect/client'
+import { SessionTypes } from '@walletconnect/types'
+import * as ethUtil from 'ethereumjs-util'
+
+/**
+ * Implements the signer interface on top of the WalletConnect interface.
+ */
+export class WalletConnectSigner implements Signer {
+  /**
+   * Construct a new instance of a WalletConnectSigner
+   */
+  constructor(
+    protected client: WalletConnect,
+    protected session: SessionTypes.Settled,
+    protected account: string
+  ) {}
+
+  async signTransaction(): Promise<{ v: number; r: Buffer; s: Buffer }> {
+    throw new Error('signTransaction unimplemented; use signRawTransaction')
+  }
+
+  async signRawTransaction(tx: CeloTx): Promise<EncodedTransaction> {
+    const result = await this.client.request({
+      topic: this.session.topic,
+      chainId: '44787',
+      request: {
+        method: 'eth_signTransaction',
+        params: tx,
+      },
+    })
+    return result
+  }
+
+  async signTypedData(data: EIP712TypedData): Promise<{ v: number; r: Buffer; s: Buffer }> {
+    const params = [this.account, JSON.stringify(data)]
+    const result = await this.client.request({
```

Comment:
There seems a fair amount of common structure that you could probably just extract